### PR TITLE
feat: Add Image Extractor

### LINKED ISSUE #10: [FEATURE] 选题时图片自动提交到图床
增加 image_extractor 模块与 image_uploader 模块，分别用于从从过滤后的文章中解析出图片文件和上传（提交）文章所需的图片。考虑增加一个本地清单文件，用于记录需要上传的图片以及上传的图片与原文图片链接的对应关系，便于后续将原始图片外链替换为 FOSScope 图床上的图片链接。

### PR DESCRIPTION
**Pull request by cysnies**
Added module "image_extractor.rs" that filters and saves images from the original article to a specified local directory based on their file extensions.

### CODE REVIEW

File: src-rust/toolkit-core/Cargo.toml [hunk 1]
```diff
@@ -19,10 +19,11 @@ serde = { version = "1", features = ["derive"] } # Serialization
 serde_json = "1.0.117" # JSON Serialization
 toml = "0.8.14" # TOML Serialization
 regex = "1.10.5" # Regular Expression
-reqwest = { version = "0.12", features = ["json"] } # HTTP Client
+reqwest = { version = "0.12", features = ["json","blocking"] } # HTTP Client
```
Comment (Qubik65536) [inline src-rust/toolkit-core/Cargo.toml]:
```suggestion
reqwest = { version = "0.12", features = ["json", "blocking"] } # HTTP Client
```

File: src-rust/toolkit-core/src/parser/image_extractor.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+use reqwest::blocking::get;
+use kuchiki::traits::*;
+use std::fs::{File, create_dir_all};
+use std::io::copy;
+use std::path::Path;
+
+/*
```
Comment (Qubik65536) [inline src-rust/toolkit-core/src/parser/image_extractor.rs:7]: 我们同步一下全部用 `///` 样式的注释吧

File: src-rust/toolkit-core/src/parser/image_extractor.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                        let mut file = File::create(target_path)?;
+                        copy(&mut response.bytes()?.as_ref(), &mut file)?;
+                    }
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
```
Comment (Qubik65536) [inline src-rust/toolkit-core/src/parser/image_extractor.rs:41]: 结尾加空行

**Review by Qubik65536 — CHANGES_REQUESTED** [review]

File: src-rust/toolkit-core/src/parser/image_extractor.rs (hunk 2 above)
Comment (cysnies) [reply src-rust/toolkit-core/src/parser/image_extractor.rs:7]: OK

**Pull request closed without merging**